website/blog/2024-08-07-manipulate-devtools.md

```diff
@@ -0,0 +1,325 @@
+You can find a complete example in [Github](https://github.com/gromanas/wdio-devtools-demo)
+
+# Devtools in WebdriverIO v8
+
+## Setting Up Extension
+
+You need to download the extension you want to test in a `.crx` format,
+in our case we have downloaded the `PixiJS-Devtools-Chrome-Web-Store.crx` extension file.
+
+## Setting Up WebDriverIO
+
+First, ensure you have WebDriverIO set up in your project. If not, you can install it by following the official installation guide [here](https://webdriver.io/docs/gettingstarted#initiate-a-webdriverio-setup)
+
+## WebDriverIO Configuration for Chrome DevTools
+
+Here is the configuration file (`wdio.conf.js`) tailored to our example:
+
+```javascript
+import path from "path";
+import fs from "fs";
+
+export const config = {
+  specs: ["./test/specs/**/*.js"],
+  runner: "local",
+  maxInstances: 1,
+  capabilities: [
+    {
+      browserName: "chrome",
+      browserVersion: "stable",
+      "goog:chromeOptions": {
+        args: [
+          "ignore-certificate-errors-spki-list",
+          "--ignore-certificate-errors",
+          "window-size=1920,1080",
+          "--auto-open-devtools-for-tabs",
+        ],
+        extensions: [
+          fs.readFileSync(
+            path.resolve("./PixiJS-Devtools-Chrome-Web-Store.crx"),
+            "base64"
+          ),
+        ],
+      },
+      "goog:loggingPrefs": {
+        browser: "INFO",
+      },
+    },
+  ],
+  logLevel: "debug",
+  services: [["devtools"]],
+  framework: "mocha",
+  mochaOpts: {
+    ui: "bdd",
+    timeout: 1000000,
+  },
+};
+```
+
+In this configuration, we specify the Chrome browser with options to auto-open DevTools for tabs and load the PixiJS DevTools extension. The `extensions` option reads the extension file in base64 format.
+We need also the `--auto-open-devtools-for-tabs` flag to automatically open the DevTools (Developer Tools) panel.
+
+## Writing the Test Script
+
+Next, we write a test script to interact with the PixiJS extension through Chrome DevTools. Create a test file, for example, `devtools.test.js`, and add the following code:
+
+```javascript
+import { $, browser } from "@wdio/globals";
+
+describe("DevTools Test on latest chrome", function () {
+  it("Devtools Navigation", async function () {
+    // Open an example game.
+    await browser.url(
+      "https://pixijs.io/examples-v5/#/demos-basic/container.js"
+    );
+
+    // Get window handles.
+    const handles = await browser.getWindowHandles();
+
+    // Switch to devtools window.
+    await browser.switchToWindow(handles[1]);
+
+    // Navigate through devtool tabs (for Linux use ['Control', '[']).
+    await browser.keys(["Meta", "["]);
```

Comment:
Can we use 

```ts
import { Key } from 'webdriverio';

await browser.keys([Key.Meta, "["]);
```

instead , as this is the recommended way to use special keys.

Reply:
done